packages/ui/components/topbanner/TopBanner.tsx

```diff
@@ -0,0 +1,48 @@
+import { ExclamationIcon } from "@heroicons/react/outline";
+import { XIcon } from "@heroicons/react/solid";
+import classNames from "classnames";
+import { ReactNode, useState } from "react";
+
+export type TopBannerProps = {
+  text: string;
+  variant?: keyof typeof variantClassName;
+  actions?: ReactNode;
+};
+
+const variantClassName = {
+  default: "bg-gradient-primary",
+  warning: "bg-orange-400",
+  error: "bg-red-400",
+};
+
+export function TopBanner(props: TopBannerProps) {
+  const { variant = "default", text } = props;
+  const [isOpen, setOpen] = useState(true);
```

Comment:
I know this is still a draft and WIP but I'll quickly leave me concern here. 
Using internal state on this component will mean we also need to create a `useState` in the parent component to toggle the functionality. 

My thoughts would be to take in `isOpen & setOpen` from props. This would allow us to conditionally render the component rather than use css to show/hide it.

Reply:
yeah @sean-brydon  we should take isOpen from props and maybe we create a context wrapper so that we can invoke setOpen from anywhere.

Reply:
Good shout :)